src/components/Product.js

```diff
@@ -43,9 +45,16 @@ function Product({ product, baseUnit, totalVolume }) {
         Total Volume in {baseUnit}: {totalVolume}
       </p>
       <label htmlFor={id} className={styles.priceLabel}>Price: $
-        <input id={id} type="text" value={price} onChange={(event) => {
-          handlePriceChange(event.target.value)
-        }} />
+        <input
+          id={id}
+          type="text"
+          value={price}
+          onChange={(event) => {
+            handlePriceChange(event.target.value)
+          }}
+          onBlur={(event) => {
+            updateItemPrice(currentCollectionId, product.id, event.target.value)
+          }} />
```

Comment:
@robtarr if you can take another look at this, I think using `onBlur` as you suggested makes sense and is working now to update collections without `useRef`.

I'm still seeing every product in the grid re-render on blur (see console logs), so I think my next PR will be to add memoization for products and only re-render if their individual price changes. Let me know what you think.